smdebug/tensorflow/base_hook.py

```diff
@@ -489,12 +491,14 @@ def set_optimizer_variables(self, optimizer_variables):
         # From TF 2.2, executing_eagerly_outside_functions() can be used as
         # ops.executing_eagerly_outside_functions() or tf.compat.v1.executing_eagerly_outside_functions().
         # But in TF 2.1, only ops.executing_eagerly_outside_functions() is valid
-        if is_tf_version_2x() and ops.executing_eagerly_outside_functions():
-            return
         # since this is done for each variable at a time for keras, not checking if set already
         self.collection_manager.get(CollectionKeys.OPTIMIZER_VARIABLES).add_for_mode(
             optimizer_variables, ModeKeys.TRAIN
         )
+        if self.save_all is True:
+            self.collection_manager.get(CollectionKeys.ALL).add_for_mode(
```

Comment:
how has it been adding to "all" until now? why dint we need this check so far?

Reply:
See comment above.